FEATURE: Support for CSS-like id and class selector

jonnitto opened the pull request:
In some enviroments, it is not as easy to get the id selector correctly written. This add the support for writing the selector for an id in a simple CSS alike way

jonnitto committed:
FEATURE: Support for CSS-like id selector

In some enviroments, it is not as easy to get the id selector correctly written. This add the support for writing the selector for an id in a simple CSS alike way

jonnitto committed:
TASK: Remove empty space on readme

mficzel commented:
I dislike the switch between xpath and selectors but get your reasoning?

how about a breaking release (major) and switch entirely to css selectors? In all cases I know we always streamed to the head anyways and never changed the default.

jonnitto commented:
If we switch entirely, we would have to include a dependency or a lot more code. Or we just say, you can select `id` or tags…

jonnitto committed:
TASK: Update readme

jonnitto committed:
Merge branch 'main' into feature/id_slector

jonnitto committed:
Fix: Tag mode and explain xPath mode

mficzel committed:
Merge branch 'main' into feature/id_slector

mficzel reviewed (changes requested):
I agree to this when a Testcase is added to the SlipstreamServiceTest

jonnitto committed:
Update: Add classname selector and don't allow spaces on id selectors

jonnitto committed:
Update: Add tests for id and class name selector

jonnitto committed:
Docs: Rename section from "Prepend a tag" to "Location of the tag"

jonnitto committed:
Update: Test also if the target gets trimmed

jonnitto committed:
Update: Add tests for invalid class name and id selectors

jonnitto committed:
Update: Add tests for after and before location processing in SlipStreamService

jonnitto commented:
I removed the automatic tag selector, as it can happen someone uses a complex XPATH. But I added a class name selector, as this is kind of complicated to write. I also improve the id selector and explained it in the readme.

I also add tests for after and before location processing in SlipStreamService

jonnitto committed:
Docs: Improve clarity in CSS selector usage section

jonnitto committed:
Docs: Remove whitespace in code block and fix typo
